core/trino-main/src/test/java/io/trino/sql/gen/BenchmarkDynamicPageFilter.java

```diff
@@ -0,0 +1,246 @@
+        INT64_RANDOM(BIGINT, (block, r) -> BIGINT.writeLong(block, r.nextLong())),
+        INT64_FIXED_32K(BIGINT, (block, r) -> BIGINT.writeLong(block, r.nextLong() % 32768)), // LongBitSetFilter
+        /**/;
+
+        private final Type type;
+        private final ValueWriter valueWriter;
+
+        DataSet(Type type, ValueWriter valueWriter)
+        {
+            this.type = requireNonNull(type, "type is null");
+            this.valueWriter = requireNonNull(valueWriter, "valueWriter is null");
+        }
+
+        public TupleDomain<ColumnHandle> createFilterTupleDomain(int filterSize, boolean nullsAllowed)
+        {
+            List<Page> filterValues = createSingleColumnData(valueWriter, type, 0, filterSize);
+            ImmutableList.Builder<Object> valuesBuilder = ImmutableList.builder();
+            for (Page page : filterValues) {
+                Block block = page.getBlock(0).getLoadedBlock();
+                for (int position = 0; position < block.getPositionCount(); position++) {
+                    valuesBuilder.add(readNativeValue(type, block, position));
+                }
+            }
+            return TupleDomain.withColumnDomains(ImmutableMap.of(
+                    new TestingColumnHandle("dummy"),
+                    Domain.create(ValueSet.copyOf(type, valuesBuilder.build()), nullsAllowed)));
+        }
+
+        private List<Page> createInputTestData(
+                TupleDomain<ColumnHandle> filter,
+                double inputNullChance,
+                double nonNullsSelectivity,
+                long inputRows)
+        {
+            List<Object> nonNullValues = filter.getDomains().orElseThrow()
+                    .values().stream()
+                    .flatMap(domain -> {
+                        DiscreteSet nullableDiscreteSet = domain.getNullableDiscreteSet();
+                        return nullableDiscreteSet.getNonNullValues().stream();
+                    })
+                    .collect(toImmutableList());
+
+            // pick a random value from the filter
+            return createSingleColumnData(
+                    (block, r) -> {
+                        if (isTrue(r, nonNullsSelectivity)) {
+                            // pick a random value from the filter
+                            Object value = nonNullValues.get(r.nextInt(nonNullValues.size()));
+                            writeNativeValue(type, block, value);
+                            return;
+                        }
+                        valueWriter.write(block, r);
+                    },
+                    type,
+                    inputNullChance,
+                    inputRows);
+        }
+    }
+
+    @Setup
+    public void setup()
+    {
+        setup(MAX_ROWS);
+    }
+
+    public void setup(long inputRows)
+    {
+        // Pollute the JVM profile
+        for (DataSet dataSet : ImmutableList.of(DataSet.INT32_RANDOM, DataSet.INT64_FIXED_32K, DataSet.INT64_RANDOM)) {
```

Comment:
This is not guaranteed to pollute the profile. It depends on various factors, such whether it runs just on the interpreter, C1, C2, etc. A better way to do that is to run with JMH's bulk warmup mode.

Reply:
I've made that change, but in practise I'm not finding the JMH's bulk warmup mode to be a better choice.
Using that causes a multi-fold increase in the runtime of the benchmark because it warms up each test permutation for N iterations for each benchmark, whereas what we want is to warm up each test permutation once and then warm N iterations for only the benchmark.
The other way of manually polluting profile may not be guaranteed to work but in practice I have never found it to not work.
On the other hand, the JMH's bulk warmup mode is always so time consuming that I always find myself manually editing the code to remove this and pollute the profile manually to get a JHM result in reasonable amount of time.
And running JMH remotely is not the solution to this problem, it still takes up too much time.